checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -40,28 +36,47 @@ def __init__(self, resource_types: List[str], attribute: Optional[str], value: A
         self.is_jsonpath_check = is_jsonpath_check
         self.parsed_attributes: Dict[Optional[str], Any] = {}
 
-    def run(self, graph_connector: DiGraph) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    def run(self, graph_connector: DiGraph) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]], List[Dict[str, Any]]]:
         executer = ThreadPoolExecutor()
         jobs = []
         passed_vertices: List[Dict[str, Any]] = []
         failed_vertices: List[Dict[str, Any]] = []
+        unknown_vertices: List[Dict[str, Any]] = []
         for _, data in graph_connector.nodes(data=True):
             if (not self.resource_types or data.get(CustomAttributes.RESOURCE_TYPE) in self.resource_types) \
                     and data.get(CustomAttributes.BLOCK_TYPE) in SUPPORTED_BLOCK_TYPES:
-                jobs.append(executer.submit(self._process_node, data, passed_vertices, failed_vertices))
+                jobs.append(executer.submit(
+                    self._process_node, data, passed_vertices, failed_vertices, unknown_vertices))
 
         concurrent.futures.wait(jobs)
-        return passed_vertices, failed_vertices
+        return passed_vertices, failed_vertices, unknown_vertices
+
+    def get_operation(self, vertex: Dict[str, Any]) -> Optional[bool]:
+        attr_val = vertex.get(self.attribute)   # type:ignore[arg-type]  # due to attribute can be None
+        # if this value contains an underendered variable, then we cannot evaluate value checks,
+        # and will return None (for UNKNOWN)
+        # handle edge cases in some policies that explicitly look for blank values
+        if self.is_value_attribute_check and self._is_variable_dependant(attr_val, vertex['source_']) \
+                and self.value != '':
+            return None
 
-    def get_operation(self, vertex: Dict[str, Any]) -> bool:
         if self.attribute and (self.is_jsonpath_check or re.match(WILDCARD_PATTERN, self.attribute)):
             attribute_matches = self.get_attribute_matches(vertex)
+            filtered_attribute_matches = [a for a in attribute_matches if
+                                          not self._is_variable_dependant(vertex.get(a), vertex['source_'])] \
+                if self.is_value_attribute_check and self.value != '' else attribute_matches
```

Comment:
can you change it to a normal `if` block, a bit too long line for a ternary operator for my taste 😄 
something like that
```suggestion
            filtered_attribute_matches = attribute_matches
            if self.is_value_attribute_check and self.value != '':
                filtered_attribute_matches = [
                    a for a in attribute_matches
                    if not self._is_variable_dependant(vertex.get(a), vertex['source_'])
                ]
```